src/sentry/workflow_engine/endpoints/validators.py

```diff
@@ -0,0 +1,148 @@
+        return attrs
+
+
+class NumericComparisonConditionValidator(BaseDataConditionValidator):
+
+    comparison = serializers.FloatField(
+        required=True,
+        help_text="Comparison value to be compared against value from data.",
+    )
+    result = serializers.ChoiceField(
+        choices=[
+            (DetectorPriorityLevel.HIGH, "High"),
+            (DetectorPriorityLevel.MEDIUM, "Medium"),
+            (DetectorPriorityLevel.LOW, "Low"),
+        ]
+    )
+
+    @property
+    def supported_conditions(self) -> frozenset[Condition]:
+        raise NotImplementedError
+
+    @property
+    def supported_results(self) -> frozenset[DetectorPriorityLevel]:
+        raise NotImplementedError
+
+    def validate_condition(self, value: str) -> Condition:
+        try:
+            condition = Condition(value)
+        except ValueError:
+            condition = None
+
+        if condition not in self.supported_conditions:
+            raise serializers.ValidationError(f"Unsupported condition {value}")
+        return condition
+
+    def validate_result(self, value: str) -> DetectorPriorityLevel:
+        try:
+            result = DetectorPriorityLevel(int(value))
+        except ValueError:
+            result = None
+        if result not in self.supported_results:
+            raise serializers.ValidationError("Unsupported condition result")
+        return result
+
+
+class BaseDataSourceValidator(CamelSnakeSerializer):
+    pass
+
+
+class BaseGroupTypeDetectorValidator(CamelSnakeSerializer):
+    name = serializers.CharField(
+        required=True,
+        max_length=200,
+        help_text="Name of the uptime monitor",
+    )
+    group_type = serializers.CharField()
+
+    def validate_group_type(self, value: str) -> type[GroupType]:
+        detector_type = grouptype.registry.get_by_slug(value)
+        if detector_type is None:
+            raise serializers.ValidationError("Unknown group type")
+        if detector_type.detector_validator is None:
+            raise serializers.ValidationError("Group type not compatible with detectors")
+        # TODO: Probably need to check a feature flag to decide if a given
+        # org/user is allowed to add a detector
+        return detector_type
+
+    @property
+    def data_source(self) -> BaseDataSourceValidator:
+        raise NotImplementedError
+
+    @property
+    def data_conditions(self) -> BaseDataConditionValidator:
+        raise NotImplementedError
+
+    def create(self, validated_data):
+        # TODO: Implement data source abstraction
+        # data_source_data = validated_data["data_source"]
+        # data_source = DataSource.objects.create(
+        #     data_source_data
+        # )
+        condition_data = validated_data["data_conditions"]
+        with transaction.atomic(router.db_for_write(Detector)):
+            condition_group = DataConditionGroup.objects.create(
+                logic_type=DataConditionGroup.Type.ANY,
```

Comment:
Should the logic_type be passed through the `condition` info as well? (Or should we have another validator for DataConditionGroups?) 🤔

Reply:
Yeah, this should be part of the condition group validator... Maybe that validator should contain the conditions?

Reply:
Actually - I decided when building this that the api customer doesn't really need to know about the internals of our data model. Since for detectors we can only ever have one condition group, I thought it'd be a simpler mental model that we could maybe just pass this at the detector level. But you can change it as needed